File: DSA_python/DSA07003 KIỂM TRA BIỂU THỨC SỐ HỌC.py
def check(s):
    temp = "(+-*/"
    A = list()

    for i in s:
        if temp.find(i) != -1:
            A.append(i)
        elif i == ')':
            if A[len(A)-1] == '(':
                return True
            else:
                while A[len(A)-1] != '(':
                    A.pop()
                A.pop()
    
    return False

def submit():
    s = input()

    print("Yes" if check(s) == True else "No")


test = int(input())
for i in range(test):
    submit()

File: DSA/DSA07003 KIỂM TRA BIỂU THỨC SỐ HỌC.cpp
#include <bits/stdc++.h>
using namespace std;

bool check(string s)
{
    string temp = "(+-*/";
    stack<char> A;

    for (auto &&i : s)
    {
        if (temp.find(i) != string::npos)
        {
            A.push(i);
        }
        else if (i == ')')
        {
            if (A.top() == '(')
            {
                return true;
            }
            else
            {
                while (A.top() != '(')
                {
                    A.pop();
                }
                A.pop();
            }
        }
    }
    return 0;
}

void submit()
{
    string s;
    getline(cin, s);

    check(s) ? cout << "Yes" : cout << "No";
    

    cout << endl;
}

int main()
{
    ios::sync_with_stdio(false);                                    // Speed up input/output
    cin.tie(nullptr);
    cout.tie(nullptr);

    int so_bo_test;
    cin >> so_bo_test;
    cin.ignore();
    
    while (so_bo_test--)
    {
        submit();
    }

    return 0;
}